tests/data-type.js

```diff
@@ -49,3 +55,125 @@ test('private createWithDocId() method', async (t) => {
   const read = await dataType.getByDocId(customId)
   t.is(read.docId, customId)
 })
+
+test('test validity of `createdBy` field', async (t) => {
+  const projectKey = randomBytes(32)
+  const { dataType: dt1, dataStore: ds1 } = await testenv({ projectKey })
+
+  const customId = randomBytes(8).toString('hex')
+  const obs = await dt1[kCreateWithDocId](customId, obsFixture)
+  const createdBy = crypto.discoveryKey(ds1.writerCore.key).toString('hex')
+
+  t.is(
+    obs.createdBy,
+    createdBy,
+    'createdBy should be generated from the DataStore writerCore discoveryKey'
+  )
+})
+
+test('test validity of `createdBy` field from another peer', async (t) => {
+  const projectKey = randomBytes(32)
+  const {
+    coreManager: cm1,
+    dataType: dt1,
+    dataStore: ds1,
+  } = await testenv({ projectKey })
+  const {
+    coreManager: cm2,
+    dataType: dt2,
+    dataStore: ds2,
+  } = await testenv({ projectKey })
+
+  const obs = await dt1.create(obsFixture)
+  const driveId = ds1.writerCore.key
+  const { destroy } = replicateDataStore(cm1, cm2)
+  await waitForCores(cm2, [driveId])
+  const replicatedCore = cm2.getCoreByKey(driveId)
+  await replicatedCore.update({ wait: true })
+  await replicatedCore.download({ end: replicatedCore.length }).done()
+  const replicatedObservation = await dt2.getByVersionId(obs.versionId)
+
+  const createdBy = crypto.discoveryKey(ds1.writerCore.key).toString('hex')
+  t.is(replicatedObservation.createdBy, createdBy)
+  await destroy()
+})
+
+test('test validity of `createdBy` field updated from another peer', async (t) => {
+  const projectKey = randomBytes(32)
+  const {
+    coreManager: cm1,
+    dataType: dt1,
+    dataStore: ds1,
+  } = await testenv({ projectKey })
+  const {
+    coreManager: cm2,
+    dataType: dt2,
+    dataStore: ds2,
+  } = await testenv({ projectKey })
+
+  const obs = await dt1.create(obsFixture)
+  const driveId = ds1.writerCore.key
+  const { destroy } = replicateDataStore(cm1, cm2)
+  await waitForCores(cm2, [driveId])
+  const replicatedCore = cm2.getCoreByKey(driveId)
+  await replicatedCore.update({ wait: true })
+  await replicatedCore.download({ end: replicatedCore.length }).done()
+  const replicatedObservation = await dt2.getByVersionId(obs.versionId)
+
+  /** @type {import('@mapeo/schema').ObservationValue} */
+  const newObsFixture = {
+    schemaName: 'observation',
+    refs: [{ id: randomBytes(32).toString('hex') }],
+    tags: {},
+    attachments: [],
+    metadata: {},
+  }
+  dt2.update(obs.versionId, newObsFixture)
+
```

Comment:
I'm missing here a way so that the peer 1 gets the updated observation from peer 2 - which I _think_ its necessary for this test? - 
I'm probably missing something @achou11 ?

Reply:
not sure if this test is needed. the primary things that should be tested are:

1. A created document should have a `createdBy` field that matches its creator's writer core's discovery key id. 

2.  A created document that is replicated to another core...

    1. holds the same `createdBy` as when it was created.
    2. holds the same `createdBy` when making an update to that document.
    
I think the first two tests you added are addressing these cases, respectively (`test validity of `createdBy` field` and `test validity of `createdBy` field from another peer`).

I think the case you're describing would be more necessary if we had to work with an `updatedBy` field.

Reply:
in short, I think you can include the "update" test in the `test validity of createdBy field from another peer` test, where the following occurs in that test:

1. Create doc from peer 1. Check `createdBy`
2. Replicate doc from peer 1 to peer 2.
3. Get doc in peer 2 and check `createdBy`
4. Update doc in peer 2 and checkc `createdBy`

Reply:
Ok, in the last commit I added that missing check on the second test (that `createdBy` is doesn't change when updating the document). 
I think that this is maybe ready to merge?